generator/mapper.go

```diff
@@ -0,0 +1,123 @@
+package generator
+
+import (
+	"os"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+
+	"github.com/opencontrol/oscalkit/types/oscal/catalog"
+	"github.com/opencontrol/oscalkit/types/oscal/profile"
+)
+
+//CreateCatalogsFromProfile maps profile controls to multiple catalogs
+func CreateCatalogsFromProfile(profileArg *profile.Profile) ([]*catalog.Catalog, error) {
+
+	done := 0
+	errChan := make(chan error)
+	doneChan := make(chan bool)
+	var outputCatalogs []*catalog.Catalog
```

Comment:
Add a mutex to prevent data race

```suggestion
 	var outputCatalogs []*catalog.Catalog
 	mut := &sync.Mutex{}
```